File: BW/11월1주차/더맵게.js
function solution(scoville, K) {
    scoville.sort((a, b) => a - b);
    
    let mixCount = 0;

    while (scoville.length > 1 && scoville[0] < K) {
        const first = scoville.shift();
        const second = scoville.shift();

        const newScoville = first + second * 2;

        scoville.push(newScoville);
        scoville.sort((a, b) => a - b); // 삽입 후 오름차순 정렬

        mixCount++;
    }

    // 모든 음식의 스코빌 지수가 K 이상인지 확인
    return scoville[0] >= K ? mixCount : -1;
}

File: JY/11월 2주차/더맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)  # 리스트를 최소 힙으로 변환

    while True:
        # 가장 낮은 스코빌 지수가 K 이상인지 확인
        if scoville[0] >= K:
            break

        # 스코빌 지수가 K 이상으로 만들 수 없는 경우
        if len(scoville) < 2:
            return -1

        # 가장 낮은 두 개의 요소를 꺼내어 새로운 스코빌 지수를 계산
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        new_scoville = first + (second * 2)
        heapq.heappush(scoville, new_scoville)
        
        answer += 1

    return answer
